### PR TITLE
DrivAerML: metric-aware w=0.05 + lr=4.5e-4 (narrow LR probe)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Test a slightly lower learning rate (4.5e-4 vs champion 5e-4) with metric-aware loss. On MSE-only, lr=4.5e-4 gave 4.134% — a steep cliff below 5e-4. But the metric-aware auxiliary loss may change the LR optimum: the rel-L2 gradient component at 5e-4 may overshoot, and 4.5e-4 could provide a better balance.

## DM Baseline: val=3.700% (lr=5e-4, w=0.05), full-eval TEST=4.218%
**MSE-only at lr=4.5e-4:** 4.134% — steep cliff from 3.833% at lr=5e-4

## Instructions
```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --dm-loss mse_plus_raw_rel_l2 --dm-rel-l2-weight 0.05 \
  --optimizer adamw --lr 4.5e-4 --cosine-t-max 30 --grad-clip 0.5 \
  --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint \
  --eval-interval 10 --seed 42 --wandb-group dm-metric-aware-sweep \
  --kill-if-best-val-above 150:8.0 --kill-if-no-improvement 200:0.05
```
**Target:** val < 3.700%

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT taki:

## Results

### DrivAerML: metric-aware w=0.05 + lr=4.5e-4 (narrow LR probe)

| Metric | Baseline (lr=5e-4, w=0.05) | MSE-only (lr=4.5e-4) | This run (lr=4.5e-4, w=0.05) |
|--------|---------------------------|----------------------|------------------------------|
| val_primary/surface_rel_l2_pct | **3.700%** | 4.134% | 3.787% |
| test_primary/surface_rel_l2_pct | 4.218% | — | 4.357% |

- **Best val:** 3.787% at epoch 900 (target was < 3.700%)
- **Best test (from best checkpoint):** 4.357%
- **Final val (ep999):** 3.792%
- **Epochs:** 999 (hit epoch limit)
- **Best epoch:** 900

### Command
```bash
CUDA_VISIBLE_DEVICES=0 SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --dm-loss mse_plus_raw_rel_l2 --dm-rel-l2-weight 0.05 \
  --optimizer adamw --lr 4.5e-4 --cosine-t-max 30 --grad-clip 0.5 \
  --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint \
  --eval-interval 10 --seed 42 --wandb-group dm-metric-aware-sweep \
  --kill-if-best-val-above 150:8.0 --kill-if-no-improvement 200:0.05 \
  --agent taki --wandb-name "taki/dm-metric-aware-w005-lr45e4"
```

### W&B
- Run ID: `dj9zz7dk`
- URL: https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/dj9zz7dk

### What happened

The metric-aware loss (`mse_plus_raw_rel_l2`, w=0.05) at lr=4.5e-4 **significantly improves over MSE-only at the same LR** (3.787% vs 4.134%, a 0.347% improvement), confirming that the metric-aware auxiliary loss shifts the LR optimum. However, it **did not match the champion** at lr=5e-4 (3.787% vs 3.700%, 0.087% worse on val; 4.357% vs 4.218%, 0.139% worse on test).

**Convergence trajectory:** The model showed steady improvement through 999 epochs with a clear cosine-cycle-driven pattern (T_max=30). Major gains came at cycle troughs (every ~30 epochs), with near-flat intervals between. The val metric went from 42% (ep10) → 7.1% (ep100) → 4.1% (ep450) → 3.79% (ep900). The training loss continued dropping throughout but val showed increasing overfitting signals after ep500 (training loss dropping while val plateaued for 50-100 epoch stretches).

**Key observations:**
1. The metric-aware loss rescues lr=4.5e-4 from the "steep cliff" seen with MSE-only (4.134% → 3.787%), narrowing the gap to the 5e-4 champion from 0.43% to 0.087%.
2. Despite running the full 999 epochs, the model never reached the 3.700% target — convergence slowed to ~0.02% per cosine cycle in the final 200 epochs.
3. The val-test gap is larger than baseline (0.57% vs 0.52%), suggesting slightly more overfitting at the lower LR, possibly because the model takes more epochs to converge and thus sees the training data more times.

### Suggested follow-ups
- **lr=4.75e-4 with metric-aware loss:** Split the difference between 4.5e-4 and 5e-4 to see if there's a sweet spot in between.
- **Larger cosine T_max at lower LR:** With lr=4.5e-4 needing 999 epochs, a longer T_max (e.g., 50) might give the model more time in productive LR ranges per cycle.
- **This confirms lr=5e-4 is the right LR for metric-aware loss.** The cliff that existed with MSE-only is substantially softened by the auxiliary loss, but the optimum hasn't shifted enough to favor 4.5e-4.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3350 because ADVISOR: Closing — val=3.787% with w=0.05+lr=4.5e-4 at T_max=30. Well below 3.622% T_max=36 baseline. lr=4.5e-4 at T_max=30 converges too slowly. Reassigning to lr=4.8e-4+T_max=36 compound..
